jni/src/linear-types/Foreign/JNI/Safe.hs

```diff
@@ -100,8 +100,11 @@ throwNew :: MonadIO m => JNI.JClass -> JNI.String ->. m ()
 throwNew jclass = Unsafe.toLinear $ \msg ->
     liftIO (JNI.throwNew jclass msg)
 
-findClass :: MonadIO m => ReferenceTypeName -> m (Unrestricted JNI.JClass)
-findClass name = liftIO (Unrestricted <$> JNI.findClass name)
+findClass
+  :: MonadIO m
+  => ReferenceTypeName
+  -> m (UnsafeUnrestrictedReference JNI.JClass)
```

Comment:
Tangentially to this PR, it is looking to me like we could just use linear class references.

If the safe API is too cumbersome to use when class references are returned, then we could just delete them instead of returning them.